### PR TITLE
Fix mask combinator move from false (GEN-365)

### PR DESCRIPTION
**Pull request by sritchie**
We also added a test for the case from True => True.

### CODE REVIEW

File: src/genjax/_src/generative_functions/combinators/mask.py [hunk 1]
```diff
@@ -134,62 +133,76 @@ def update_change_target(
...
+            final_trace = premasked_trace
+        else:
+            final_trace: Trace[R] = jtu.tree_map(
+                lambda v1, v2: jnp.where(check_arg, v1, v2),
+                premasked_trace,
+                inner_trace,
+            )
+
+        weight = (
+            FlagOp.and_(check, check_arg) * weight
```
Comment (sritchie) [inline src/genjax/_src/generative_functions/combinators/mask.py]: commenting this out doesn't affect the tests, since the only test we have has `0.0` for weight.

File: src/genjax/_src/generative_functions/combinators/mask.py [hunk 2]
```diff
@@ -134,62 +133,76 @@ def update_change_target(
...
+            + (FlagOp.and_(check, FlagOp.not_(check_arg))) * (-inner_trace.get_score())
+            + FlagOp.and_(FlagOp.not_(check), FlagOp.not_(check_arg)) * 0.0
```
Comment (sritchie) [inline src/genjax/_src/generative_functions/combinators/mask.py]: why do we have this line in here? Can we leave the math comment and delete, given that we multiply by 0.0?

File: src/genjax/_src/generative_functions/combinators/mask.py (hunk 1 above)
Comment (MathieuHuot) [reply src/genjax/_src/generative_functions/combinators/mask.py]: Why do we need to distinguish the case of the inner_trace being an empty trace?

Comment (sritchie) [reply src/genjax/_src/generative_functions/combinators/mask.py]: @MathieuHuot if we don't, there's a test that fails when it attempts to `tree_map` a `MaskTrace` and an `EmptyTrace` together.

**Review by femtomc — APPROVED** [review]

**Review by MathieuHuot — APPROVED** [review]
